test/e2e/watchcache/watchcache_enabled_test.go

```diff
@@ -202,3 +214,60 @@ func collectCacheHitsFor(ctx context.Context, t *testing.T, config *rest.Config,
 	}
 	return totalCacheHits, prefixCacheHit
 }
+
+const resyncPeriod = 10 * time.Hour
+const byWorkspace = "byWorkspace"
+
+func testDynamicDiscoverySharedInformerFactory(t *testing.T, cfg *rest.Config, expectedGVR schema.GroupVersionResource, expectedResName string, expectedClusterName logicalcluster.Name) {
+	t.Logf("Starting DynamicDiscoverySharedInformerFactory and checking if it has %v with name %v in cluster %v", expectedGVR.String(), expectedResName, expectedClusterName)
```

Comment:
nit: move "Checking if ..." down above the `require.Eventually`. It's important to have a flow of `t.Logs` in e2e to understand where something breaks.